tests/bluetooth/audio/cap_commander/src/test_broadcast_reception.c

```diff
@@ -0,0 +1,442 @@
+/* test_broadcast_reception.c - unit test for broadcast reception */
+
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+
+#include <zephyr/bluetooth/audio/cap.h>
+#include <zephyr/bluetooth/audio/vcp.h>
+#include <zephyr/fff.h>
+
+#include "bluetooth.h"
+#include "cap_commander.h"
+#include "conn.h"
+#include "expects_util.h"
+#include "cap_mocks.h"
+#include "test_common.h"
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(bt_broadcast_reception_test, CONFIG_BT_CAP_COMMANDER_LOG_LEVEL);
+
+#define FFF_GLOBALS
+
+#define ADDRESS 0x123456
+#define SID          0x0E
+#define ADV_INTERVAL 10
+#define BROADCAST_ID 0x55AA55
+#define NR_SUBGROUPS 2
+
+struct cap_commander_test_broadcast_reception_fixture {
+	struct bt_conn conns[CONFIG_BT_MAX_CONN];
+
+	struct bt_bap_bass_subgroup start_subgroups[NR_SUBGROUPS];
+	struct bt_cap_commander_broadcast_reception_start_member_param
+		start_member_params[CONFIG_BT_MAX_CONN];
+	struct bt_cap_commander_broadcast_reception_start_param start_param;
+};
+
+static void cap_commander_test_broadcast_reception_fixture_init(
+	struct cap_commander_test_broadcast_reception_fixture *fixture)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(fixture->conns); i++) {
+		test_conn_init(&fixture->conns[i]);
+	}
+}
+
+static void *cap_commander_test_broadcast_reception_setup(void)
+{
+	struct cap_commander_test_broadcast_reception_fixture *fixture;
+
+	fixture = malloc(sizeof(*fixture));
+	zassert_not_null(fixture);
+
+	return fixture;
+}
+
+static void cap_commander_test_broadcast_reception_before(void *f)
+{
+	memset(f, 0, sizeof(struct cap_commander_test_broadcast_reception_fixture));
+	cap_commander_test_broadcast_reception_fixture_init(f);
+}
+
+static void cap_commander_test_broadcast_reception_after(void *f)
+{
+	struct cap_commander_test_broadcast_reception_fixture *fixture = f;
+
+	bt_cap_commander_unregister_cb(&mock_cap_commander_cb);
+
+	for (size_t i = 0; i < ARRAY_SIZE(fixture->conns); i++) {
+		mock_bt_conn_disconnected(&fixture->conns[i], BT_HCI_ERR_REMOTE_USER_TERM_CONN);
+	}
+}
+
+static void cap_commander_test_broadcast_reception_teardown(void *f)
+{
+	free(f);
+}
+
+static void init_start_param(void *f)
+{
+	struct cap_commander_test_broadcast_reception_fixture *fixture = f;
+	int err;
+
+	fixture->start_param.type = BT_CAP_SET_TYPE_AD_HOC;
+	fixture->start_param.param = fixture->start_member_params;
+
+	fixture->start_param.count = ARRAY_SIZE(fixture->start_member_params);
+
+	for (size_t i = 0; i < ARRAY_SIZE(fixture->start_subgroups); i++) {
+		fixture->start_subgroups[i].bis_sync = 1 << i;
+#if defined(CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_SIZE)
+		/* TODO: add tests where metadata_len > 0 */
+		fixture->start_subgroups[i].metadata_len = 0;
+#else
+		fixture->start_subgroups[i].metadata_len = 0;
+#endif /* CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_SIZE */
```

Comment:
The `if` and `else` clauses are identical

Reply:
For now, yes, I do really want to add a test where metadata_len >  0

Reply:
If you are adding it in this PR, then that's fine. Otherwise it doesn't really make that much sense, but not going to block your PR on this :)